core/src/main/java/org/elasticsearch/index/query/MoreLikeThisQueryBuilder.java

```diff
@@ -326,6 +384,48 @@ public final String toString() {
             }
         }
 
+        public QueryValidationException validate() {
+            QueryValidationException validationException = null;
+            if (id == null && doc == null) {
+                validationException = new QueryValidationException();
+                validationException.addValidationError("item must either have an '_id' or be an artificial 'doc'");
+            }
+            if (id != null && doc != null) {
+                validationException = validationException == null ? new QueryValidationException() : validationException;
+                validationException.addValidationError("either 'id' or 'doc' can be specified, but not both!");
+            }
+            return validationException;
+        }
+
+        @Override
+        public Item readFrom(StreamInput in) throws IOException {
+            Item item = new Item(in.readOptionalString(), in.readOptionalString(), in.readOptionalString());
```

Comment:
This will need to change to read a boolean to tell it whether it needs to read a doc or Id so it can use the right constructor and doesn't trip up passing nulls into it